File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <unordered_map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    unordered_map<string, int> types;
    
    for (int i = 0; i < clothes.size(); i++) {
        types[clothes[i][1]]++;
    }
    for (const auto& [key, value] : types) {
        answer *= (value + 1);
    }
    answer--;
    
    
    return answer;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    d = dict()
    for cloth in clothes:
        if cloth[1] in d:
            d[cloth[1]] += 1
        else:
            d[cloth[1]] = 1
            
    for num in d.values():
        answer *= num+1
    
    return answer-1